components/config-mgmt-service/integration_test/node_run_test.go

```diff
@@ -61,13 +70,170 @@ func TestNodeRunWithRuns(t *testing.T) {
 		})
 }
 
+func TestNodeRunWithRunsNonmatchingNode(t *testing.T) {
+	var (
+		nodeID = newUUID()
+		run    = newIngestRun(nodeID, "success",
+			time.Now().Format(time.RFC3339),
+			time.Now().Add(time.Minute).Format(time.RFC3339))
+		req  = request.NodeRun{NodeId: run.EntityUuid, RunId: run.RunID}
+		ctx  = context.Background()
+		node = iBackend.Node{
+			Exists: true,
+			NodeInfo: iBackend.NodeInfo{
+				EntityUuid: "not-matching-run",
+			},
+		}
+	)
+
+	suite.IngestNodes([]iBackend.Node{node})
+	suite.IngestRuns([]iBackend.Run{run})
+	defer suite.DeleteAllDocuments()
+
+	_, err := cfgmgmt.GetNodeRun(ctx, &req)
+	assert.Error(t, err)
```

Comment:
Would it be worth making this test that the first return value is nil?

Reply:
I was under the impression that the golang way was that if an error was returned the other return value could be anything. If this is the case then I don't think we should test the second value.

Reply:
Hmm, it seems like that would allow undefined behavior, where some half-baked data could leak through because we're not explicit about our error cases.

Reply:
Does it matter if we have half-baked data when we return an error? I thought when an error was returned the other value should not be used.

Reply:
It shouldn't, but we can always test to make sure it isn't. ;)